### ISSUE TITLE
Missing values in custom mapping list

### ISSUE THREAD

**Issue by ppawlak** [metabase/metabase#5782]
- Your browser and the version: Chrome 60
- Your operating system: Windows 10
- Your databases: Postgres 9.4
- Metabase version: 0.25.2
- Metabase hosting environment:  Linux/Ubuntu 16.10
- Metabase internal database: Postgres 9.4

The "original values" I can map label to are incompelete.
See on this screenshot there are only 4 values:

![metabase ponytech net-admin-datamodel-database-5-table-127-711](https://user-images.githubusercontent.com/1941270/29508114-87c9d0c0-8653-11e7-8e4a-a17a8bb359fc.png)

whereas when I query the table in question I got 8:

```
# select distinct(status) from newsletter_contactmailingstatus;
 status
--------
      6
     -1
      4
      5
      1
      2
      0
      7
(8 rows)

```
I think the problem is not new, I had it before updating to Metabase 0.25, when making a question and filtering on this column's values.

Let me know if you need more details to reproduce this issue.

**Comment by mazameli** [comment]
Hmm, this could be happening if you've turned "in-depth analysis" off, because then Metabase wouldn't have a cached list of distinct values present in that field. [More information here](https://github.com/metabase/metabase/issues/5680)

**Comment by ppawlak** [comment]
"in-depth analysis" is turned on on this database.

**Comment by salsakran** [comment]
This should be alleviated once an RC is cut with the new sync code. In the meantime, you should be able to re-sync the DB and get all of those.

**Comment by ppawlak** [comment]
Sync-ing DB again did not fix the problem, I only have 4 values displayed but different ones, and with wrong labels : 

![metabase ponytech net-admin-datamodel-database-5-table-127-711](https://user-images.githubusercontent.com/1941270/29659449-9b6def72-88be-11e7-9807-f705429f80c2.png)

**Comment by senior** [comment]
@ppawiak The problem you're seeing is caused by changes in field values. This can lead to having the remappings incorrect or missing and there's not a way to fix it in the UI. I just pushed up a PR that does that reconciling. I'm hoping that this can be included in the next 0.26 RC release.

**Comment by ppawlak** [comment]
@senior Thanks for the follow up, I'm looking forward the 0.26 release :)

In the meantime I found a workaround to use the missing values in questions filter: I use _between_  (and put the same value twice) instead of _equal_.
